source/guides/clear/swupd-3rd-party.rst

```diff
@@ -0,0 +1,284 @@
+
+   .. code-block:: bash
+      
+      sudo mixer build bundles
+      sudo mixer build update
+
+Set up a web server to host the bundles   
+***************************************
+
+Now, we will install a web server, ngnix for this example,
+to host the bundles so clients can access them.  
+
+#. Install the `nginx` bundle.
+
+   .. code-block:: bash
+ 
+      sudo swupd bundle-add nginx
+
+#. Configure and start nginx.  Below is an example configuration:
+
+   .. code-block:: bash
+
+      # create symlink to mixer update directory
+      sudo mkdir -p /var/www
+      sudo ln -sf $HOME/mixer/update/www /var/www/mixer
+
+      # nginx configuration files
+      sudo mkdir -p  /etc/nginx/conf.d
+      sudo cp -f /usr/share/nginx/conf/nginx.conf.example /etc/nginx/nginx.conf
+      
+      sudo tee -a /etc/nginx/conf.d/my-web-server.conf << EOF
+      server {
+        server_name localhost;
+        location / {
+          root /var/www/mixer;
+          autoindex on;
+        }
+      }
+      EOF
+
+      # grant $USER permission to run the web server
+      sudo tee -a /etc/nginx/nginx.conf << EOF
+      user $USER;
+      EOF
+
+      # set it to auto start on boot and start it
+      sudo systemctl daemon-reload
+      sudo systemctl enable nginx --now 
+
+#. Verify you can access the website by going to http://<IP-address-of-web-server>
+
+Install and manage 3rd-party bundle on client system
+****************************************************
+
+Finally, we will use the :command:`swupd` client tool to install and 
+manage the bundle we created with :command:`mixer` earlier.
+All installed 3rd-party bundle reside in :file:`/opt`.
+
+#. First, add a repo link to the web server.
+   The `os-core` bundle will be added automatically when adding a repo.  
+   It contains items that mixer injects into the mix such as version information, 
+   format, CONTENTURL, VERSIONURL, and certificate.
+    
+   .. code-block:: bash
+
+      sudo swupd 3rd-party add my-3rd-party-repo \
+      http://<IP-address-of-web-server> --allow-insecure-http
+
+   .. note::
+      
+      By default, the :command:`swupd` client is designed to communicate 
+      with an HTTPS server. For development purposes, the swupd client 
+      can talk to an HTTP server if you add the flag ``--allow-insecure-http``.
+
+      To avoid adding this flag each time when invoking :command:`swupd`, enter:
+
+      .. code-block:: bash
+
+         mkdir -p /etc/swupd
+
+         cat > /etc/swupd/config << EOF
+         [GLOBAL]
+         allow-insecure-http=true
+         EOF
+
+#. Query the list of bundles from the repo.
+   
+   .. code-block:: bash
+
+      sudo swupd 3rd-party bundle-list -a 
+
+#. Add the helloclear bundle.
+   
+   .. code-block:: bash
+
+   sudo swupd 3rd-party bundle-add helloclear
+
+#. List installed 3rd-party bundles.
+
+   .. code-block:: bash
+
+      sudo swupd 3rd-party bundle-list
+
+#. Look in :file:`/opt` to confirm they were installed there.  
```

Comment:
It would be a good bit of info to include in the intro that 3rd-party bundles are ultimately installed under `/opt/3rd-party`. This might resonate with certain readers faster.

Reply:
I put this in the "Install and manage 3rd-party bundle on client system" section.